libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -411,6 +414,14 @@ export class SkyDateRangePickerComponent
   public validate(control: AbstractControl): ValidationErrors | null {
     if (!this.#control) {
       this.#control = control;
+
+      this.#updateRequiredStates();
+
+      this.#control.statusChanges
+        ?.pipe(takeUntil(this.#ngUnsubscribe))
+        .subscribe(() => {
```

Comment:
Subscribing in the `validate` method is a bit dangerous since it gets called every time the field's validations trigger.

Reply:
This unfortunately is the only thing we can watch to then react if validators are dynamically added or removed. We do the same thing in text editor. I wish Angular had a `validatorChanges` event - but it doesn't :(

Reply:
It's safe to subscribe to the `statusChanges` event, just not in the `validate` method. 🖖